From 4bb134c7df00daf59f3de80c8c2e09b26ed359d0 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 25 May 2021 13:53:37 +0300
Subject: [PATCH] [SW-43147] i2cslave: support module power command

Adding support for command code '100' which returns the module
power state.

Change-Id: I280185a6bb55c63e19fed711eb750dd9d1b38dc3
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 942c2ca..1d420ff 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -118,6 +118,9 @@
 	case HL_INDIRECT_CTRL_STS_DATA_3:
 		ret = i2csi_ctrl_handle_status(cmd);
 		break;
+	case HL_MODULE_POWER_STATE:
+		ret = HL_MODULE_POWER_FULL;
+		break;
 	case HL_MODULE_I2C_FREQ:
 		ret = HL_SLAVE_FREQ_400KHZ;
 		break;
